packages/broker/src/plugins/websocket/WebsocketServer.ts

```diff
@@ -85,11 +90,18 @@ export class WebsocketServer {
         })
 
         this.wss.on('connection', (ws: WebSocket, _request: http.IncomingMessage, connection: Connection) => {
-            connection.init(ws, this.streamrClient, payloadFormat)
-            addPingListener(ws)
-            if (this.pingSendInterval !== 0) {
-                addPingSender(ws, this.pingSendInterval, this.disconnectTimeout)
-            }
+            const socketId = randomString(5)
+            logger.info('Accept connection', { socketId })
+            connection.init(ws, socketId, this.streamrClient, payloadFormat).then(() => {
+                addPingListener(ws)
+                if (this.pingSendInterval !== 0) {
+                    addPingSender(ws, socketId, this.pingSendInterval, this.disconnectTimeout)
+                }
+                return
+            }, (err) => {
+                logger.warn('Close connection', { socketId, reason: err?.message })
+                ws.close()
+            })
```

Comment:
The return statement at line `100` is not needed?

Maybe `async await` and `setImmediate` would make the flow more straightforward:
```suggestion
            setImmediate(async () => {
                try {
                    await connection.init(ws, socketId, this.streamrClient, payloadFormat)
                } catch (err) {
                    logger.warn('Close connection', { socketId, reason: err?.message })
                    ws.close()
                    return
                }
                addPingListener(ws)
                if (this.pingSendInterval !== 0) {
                    addPingSender(ws, socketId, this.pingSendInterval, this.disconnectTimeout)
                }
            })
```

Reply:
Fixed in e579e9d0403b351ad93afb490afde67d55ba40b4 but without setImmediate.